scripts/warm_pg_embedded_cache_test.py

```
@@ -0,0 +1,432 @@
+# The test below validates main() by stubbing curl with a local fixture.
+
+
+def test_main_warms_cache_from_local_fixtures(
+    tmp_path: Path, curl_stub: Path
+) -> None:
+    """main() installs a complete cache entry when given a stubbed curl."""
+
+    fixture_dir = tmp_path / "fixtures"
+    fixture_dir.mkdir()
+    version = "16.10.0"
+    triple = subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)